README.md

```diff
@@ -45,44 +45,180 @@ pip install git+https://github.com/arangoml/dgl-adapter.git
+        },
+        # 3) You can specify set of strings if you want to preserve the same PyG attribute names for the node/edge type
+        "game": {"features"} # this is equivalent to {"features": "features"}
+    },
+    "edgeTypes": {
+        ("user", "plays", "game"): {
+            # 4) you can specify a list of strings for tensor dissasembly (if you know the number of node/edge features in advance)
+            "features": ["hours_played", "is_satisfied_with_game"]
+        },
+    },
+}
+
+
+adb_g = adbdgl_adapter.dgl_to_arangodb("FakeHetero", fake_hetero, metagraph, explicit_metagraph=False)
+
+# 1.3: DGL to ArangoDB with the same (optional) metagraph, but with `explicit_metagraph=True`
+# With `explicit_metagraph=True`, the node & edge types omitted from the metagraph will NOT be converted to ArangoDB.
+# Only 'user', 'game', and ('user', 'plays', 'game') will be brought over (i.e 'topic', ('user', 'follows', 'user'), ... are ignored)
+adb_g = adbdgl_adapter.dgl_to_arangodb("FakeHetero", fake_hetero, metagraph, explicit_metagraph=True)
+
+# 1.4: DGL to ArangoDB with a Custom Controller  (more user-defined behavior)
+class Custom_ADBDGL_Controller(ADBDGL_Controller):
+    def _prepare_dgl_node(self, dgl_node: dict, node_type: str) -> dict:
+        """Optionally modify a DGL node object before it gets inserted into its designated ArangoDB collection.
+
+        :param dgl_node: The DGL node object to (optionally) modify.
+        :param node_type: The DGL Node Type of the node.
+        :return: The DGL Node object
+        """
+        dgl_node["foo"] = "bar"
+        return dgl_node
+
+    def _prepare_dgl_edge(self, dgl_edge: dict, edge_type: tuple) -> dict:
+        """Optionally modify a DGL edge object before it gets inserted into its designated ArangoDB collection.
+
+        :param dgl_edge: The DGL edge object to (optionally) modify.
+        :param edge_type: The Edge Type of the DGL edge. Formatted
+            as (from_collection, edge_collection, to_collection)
+        :return: The DGL Edge object
+        """
+        dgl_edge["bar"] = "foo"
+        return dgl_edge
+
+
+adb_g = ADBDGL_Adapter(db, Custom_ADBDGL_Controller()).dgl_to_arangodb("FakeHetero", fake_hetero)
+```
+
+### ArangoDB to DGL
+```py
+# Start from scratch!
+db.delete_graph("FakeHetero", drop_collections=True, ignore_missing=True)
+adbdgl_adapter.dgl_to_arangodb("FakeHetero", fake_hetero)
+
+# 2.1: ArangoDB to DGL via Graph name (does not transfer attributes)
+dgl_g = adbdgl_adapter.arangodb_graph_to_dgl("FakeHetero")
+
+# 2.2: ArangoDB to DGL via Collection names (does not transfer attributes)
+dgl_g = adbdgl_adapter.arangodb_collections_to_dgl("FakeHetero", v_cols={"user", "game"}, e_cols={"plays"})
+
+# 2.3: ArangoDB to DGL via Metagraph v1 (transfer attributes "as is", meaning they are already formatted to DGL data standards)
```

Comment:
Do we have a definition of DGL standards in the readme?

Reply:
Good catch, I will add a reference to https://docs.dgl.ai/guide/graph.html#guide-graph in the readme, and consider that as the "DGL Standard" we are referring to

I essentially am trying to convey that Use Case #2.3 assumes the following:
- the attributes are of numerical type
- the attribute values can be stored in a Tensor without issue
- the created Tensor size is identical to the number nodes/edges of that particular attribute

Hope this makes sense